site/en/articles/x-google-ignore-list/index.md

```diff
@@ -0,0 +1,139 @@
+</figure>
+
+## What is `x_google_ignoreList`?
+
+[Source maps extensions](https://bit.ly/sourcemap#heading=h.ghqpj1ytqjbm) are additional fields that store complementary information about the source map. Such fields are prefixed with `x_`.
+
+Chrome DevTools uses the `x_google_ignoreList` field (if provided), to filter out generated code and let web developers focus only on the code they author. For example, let’s look at the following source map.
+
+```js
+/* demo.js.map */
+
+{
+  "version": 3,
+  "mappings": "AAAAA, ..." 
+  "sources": [
+    "app.js",
+    "components/Button.ts",
+    "node_modules/.../framework.js",
+    "node_modules/.../library.js",
+    ...
+  ],
+  "x_google_ignoreList": [2, 3],
+  ...
+}
+```
+
+The `sources` field shows a list of original sources used by the `mappings` entry. Watch [What are source maps?](https://youtu.be/FIYkjjFYvoI) to learn how the mappings work.
+
+Given that the two files `node_modules/…/frameworks.js` and `node_modules/.../library.js` are third party scripts, you can specify the `x_google_ignoreList` field to indicate their positions in the `sources` field. Chrome DevTools will apply this information to hide frames from those ignored files. 
+
+<figure>
+  {% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/yw7xc5vBlBrCloYwFioA.png", alt="A comparison of collapsed and expanded stack trace.", width="800", height="467" %}
+  <figcaption>
+    The image shows the stack traces before and after you expand the stack trace in the Console. The ignored frames are grayed out when expanded.
+  </figcaption>
+</figure>
+
+This also applies to the [**Call Stack**](/docs/devtools/javascript/reference/#show-ignore-listed-frames) in the **Sources** panel during breakpoint debugging.  
+
+Behind the scenes, DevTools has an extra setting enabled by default: [Automatically add known third-party scripts to ignore list](/docs/devtools/settings/ignore-list/#skip-third-party). You can find it in DevTools > **Settings** > **Ignore List**.
+
+With the `x_google_ignoreList` source map field, you have the option to [hide the ignored files](/docs/devtools/javascript/reference/#hide-ignore-listed) in the **Sources** panel to focus on your code.
+
+{% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/Y4KSjl9zJQdnAhTvtnXm.png", alt="Hide ignore-listed sources.", width="800", height="449" %}
+
+## How to populate `x_google_ignoreList`
+
+The good news is frameworks like [Angular](https://angular.io/) and [Nuxt](https://nuxt.com/) already configure `x_google_ignoreList` in their source maps. Upgrade to the latest version and it works out of the box. You get stack trace improvements effortlessly.
+
+On the other hand, build tools like [Vite](https://vitejs.dev/config/server-options.html#server-sourcemapignorelist) and [Rollup](https://rollupjs.org/configuration-options/#output-sourcemapignorelist) provide settings to configure it. There is also a [webpack plugin](https://www.npmjs.com/package/devtools-ignore-webpack-plugin) for that.
```

Comment:
@mondaychen heads up, we will mention the webpack plugin you created in an article we are going to publish today! Thank you. :)